modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterManager.java

```diff
@@ -1040,6 +1050,32 @@ public RasterManager(
                 }
             }
         }
+        if (heterogeneousCRS) {
+            // If the reader is kept open (and the rasterManager doesn't change)
+            // it would be useful to cache the result of a query for
+            // a specific EPSG code on the index, so that a DB access won't be
+            // repeated while the info is in cache.
+
+            alternativeCRSCache =
+                    CacheBuilder.newBuilder()
+                            .maximumSize(50)
```

Comment:
Magic number, best to make it a system variable if not a read parameter. 
I'd also suggest to switch the default to something larger, a classic world wide satellite mosaic case is having all UTM zones (60 zones, further split into south and north) and the two polar caps, making for 122 CRSs present at the same time in the mosaic.